core/src/com/etheller/warsmash/parsers/jass/Jass2.java

```diff
@@ -1244,3908 +1039,2283 @@ public boolean call(final CUnit unit) {
+			jassProgramVisitor.getJassNativeManager().createNative("GetTriggeringTrackable", (arguments, globalScope, triggerScope) -> {
+				throw new UnsupportedOperationException("GetTriggeringTrackable not yet implemented ???");
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("GetClickedButton", (arguments, globalScope, triggerScope) -> {
+				throw new UnsupportedOperationException("GetClickedButton not yet implemented ???");
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("GetClickedDialog", (arguments, globalScope, triggerScope) -> {
+				throw new UnsupportedOperationException("GetClickedDialog not yet implemented ???");
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("GetTournamentFinishSoonTimeRemaining", (arguments, globalScope, triggerScope) -> {
+				throw new UnsupportedOperationException("GetTournamentFinishSoonTimeRemaining not yet implemented ???");
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("GetTournamentFinishNowRule", (arguments, globalScope, triggerScope) -> {
+				throw new UnsupportedOperationException("GetTournamentFinishNowRule not yet implemented ???");
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("GetTournamentFinishNowPlayer", (arguments, globalScope, triggerScope) -> {
+				throw new UnsupportedOperationException("GetTournamentFinishNowPlayer not yet implemented ???");
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("GetTournamentScore", (arguments, globalScope, triggerScope) -> {
+				throw new UnsupportedOperationException("GetTournamentScore not yet implemented ???");
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("GetSaveBasicFilename", (arguments, globalScope, triggerScope) -> {
+				throw new UnsupportedOperationException("GetSaveBasicFilename not yet implemented ???");
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("TriggerRegisterPlayerEvent", (arguments, globalScope, triggerScope) -> {
+				final Trigger whichTrigger = arguments.get(0).visit(ObjectJassValueVisitor.getInstance());
+				final CPlayerJass whichPlayer = arguments.get(1).visit(ObjectJassValueVisitor.getInstance());
+				if (whichPlayer == null) {
+					return eventType.getNullValue();
 				}
+				final JassGameEventsWar3 whichPlayerEvent = arguments.get(2).visit(ObjectJassValueVisitor.getInstance());
+				return new HandleJassValue(eventType, whichPlayer.addEvent(globalScope, whichTrigger, whichPlayerEvent));
 			});
-			jassProgramVisitor.getJassNativeManager().createNative("CreateRegion", new JassFunction() {
-				@Override
-				public JassValue call(final List<JassValue> arguments, final GlobalScope globalScope,
-						final TriggerExecutionScope triggerScope) {
-					return new HandleJassValue(regionType, new CRegion());
+			// TODO past this point things are inconsistent about ordering
+			// jassProgramVisitor.getJassNativeManager().createNative("GetCameraMargin", new JassFunction() {
```

Comment:
Is there a reason for this extra commented code or should we remove it?

Reply:
I forgot to remove the commented code between line 1687 and 1709. Yeah we should remove it.